spring-core/src/main/java/org/springframework/core/MethodParameter.java

```diff
@@ -486,7 +486,13 @@ void setParameterType(@Nullable Class<?> parameterType) {
 		if (paramType != null) {
 			return paramType;
 		}
-		paramType = ResolvableType.forMethodParameter(this, null, 1).resolve();
+		paramType = computeParameterType();
+		// In order to improve performance, the method forMethodParameter should not be used in some cases.
+		if ("java.lang.Object".equals(paramType.getCanonicalName()) &&
+				!getDeclaringClass().equals(getContainingClass()) &&
+				!"java.lang.Object".equals(getGenericParameterType().getClass().getCanonicalName())) {
```

Comment:
Condition in the _if_ block is not readable. Extracting following code to a method with a meaningful name should be more readable.

> "java.lang.Object".equals(paramType.getCanonicalName()) &&
				!getDeclaringClass().equals(getContainingClass()) &&
				!"java.lang.Object".equals(getGenericParameterType().getClass().getCanonicalName())